2005-04-26  Gwenole Beauchesne  <gbeauchesne@mandriva.com>

	Further fixing for KDE 3.4
	* libstdc++-v3/include/bits/basic_ios.h: Make all declarations
	have default visibility.

2005-02-10  Gwenole Beauchesne  <gbeauchesne@mandrakesoft.com>

	* libstdc++-v3/include/bits/allocator.h: Make all declarations
	have default visibility.
	* libstdc++-v3/include/bits/basic_string.h: Likewise.
	* libstdc++-v3/include/std/std_istream.h: Likewise.
	* libstdc++-v3/include/std/std_ostream.h: Likewise.
	* libstdc++-v3/include/std/std_fstream.h: Likewise.

--- gcc-4.1.1/libstdc++-v3/include/bits/allocator.h.visibility1	2006-02-19 11:25:56.000000000 +0100
+++ gcc-4.1.1/libstdc++-v3/include/bits/allocator.h	2006-03-30 15:29:38.000000000 +0200
@@ -51,6 +51,8 @@
 // Define the base class to std::allocator.
 #include <bits/c++allocator.h>
 
+#pragma GCC visibility push(default)
+
 namespace std
 {
   template<typename _Tp>
@@ -129,4 +131,6 @@ namespace std
 #undef __glibcxx_base_allocator
 } // namespace std
 
+#pragma GCC visibility pop
+
 #endif
--- gcc-4.1.1/libstdc++-v3/include/bits/basic_ios.h.visibility1	2006-02-19 11:25:56.000000000 +0100
+++ gcc-4.1.1/libstdc++-v3/include/bits/basic_ios.h	2006-03-30 15:29:38.000000000 +0200
@@ -43,6 +43,8 @@
 #include <bits/locale_classes.h>
 #include <bits/locale_facets.h>
 
+#pragma GCC visibility push(default)
+
 namespace std
 {
   // 27.4.5  Template class basic_ios
@@ -464,4 +466,6 @@ namespace std
 #include <bits/basic_ios.tcc>
 #endif
 
+#pragma GCC visibility pop
+
 #endif /* _BASIC_IOS_H */
--- gcc-4.1.1/libstdc++-v3/include/bits/basic_string.h.visibility1	2006-02-19 11:25:56.000000000 +0100
+++ gcc-4.1.1/libstdc++-v3/include/bits/basic_string.h	2006-03-30 15:29:38.000000000 +0200
@@ -45,6 +45,8 @@
 #include <bits/atomicity.h>
 #include <debug/debug.h>
 
+#pragma GCC visibility push(default)
+
 namespace std
 {
   /**
@@ -2442,4 +2444,6 @@ namespace std
 #endif  
 } // namespace std
 
+#pragma GCC visibility pop
+
 #endif /* _BASIC_STRING_H */
--- gcc-4.1.1/libstdc++-v3/include/std/std_istream.h.visibility1	2006-02-19 11:25:59.000000000 +0100
+++ gcc-4.1.1/libstdc++-v3/include/std/std_istream.h	2006-03-30 15:29:38.000000000 +0200
@@ -44,6 +44,8 @@
 #include <ios>
 #include <limits> // For numeric_limits
 
+#pragma GCC visibility push(default)
+
 namespace std
 {
   // [27.6.1.1] Template class basic_istream
@@ -819,4 +821,6 @@ namespace std
 # include <bits/istream.tcc>
 #endif
 
+#pragma GCC visibility pop
+
 #endif	/* _GLIBCXX_ISTREAM */
--- gcc-4.1.1/libstdc++-v3/include/std/std_ostream.h.visibility1	2006-02-19 11:25:59.000000000 +0100
+++ gcc-4.1.1/libstdc++-v3/include/std/std_ostream.h	2006-03-30 15:29:38.000000000 +0200
@@ -43,6 +43,8 @@
 
 #include <ios>
 
+#pragma GCC visibility push(default)
+
 namespace std
 {
   // [27.6.2.1] Template class basic_ostream
@@ -525,4 +527,6 @@ namespace std
 # include <bits/ostream.tcc>
 #endif
 
+#pragma GCC visibility pop
+
 #endif	/* _GLIBCXX_OSTREAM */
--- gcc-4.1.1/libstdc++-v3/include/std/std_fstream.h.visibility1	2006-02-19 11:25:59.000000000 +0100
+++ gcc-4.1.1/libstdc++-v3/include/std/std_fstream.h	2006-03-30 15:29:38.000000000 +0200
@@ -48,6 +48,8 @@
 #include <bits/basic_file.h>
 #include <bits/gthr.h>
 
+#pragma GCC visibility push(default)
+
 namespace std
 {
   // [27.8.1.1] template class basic_filebuf
@@ -782,4 +784,6 @@ namespace std
 # include <bits/fstream.tcc>
 #endif
 
+#pragma GCC visibility pop
+
 #endif /* _GLIBCXX_FSTREAM */
